lib/train/options.rb

```diff
@@ -31,9 +31,23 @@ def option(name, conf = nil, &block)
 
       def default_options
         @default_options = {} unless defined? @default_options
+        # TODO: This is hacky way to set the default options for audit log for all type of transport.
+        @default_options.merge!(default_audit_log_options)
         @default_options
       end
 
+      def default_audit_log_options
+        # TODO: What should be the default audit log location if any of the application using train does not set it?
+        # should we keep it to $stdout.
+        {
+          enable_audit_log: { default: false },
+          audit_log_location: { required: true, default: nil },
```

Comment:
OK, yeah, that required: true broke all the tests. We can compromise.

How about set it to a path like ~/.chef/logs/train-audit.log - Use a peculiar train-specific default, and any reasonable application would have to override it. Does that work?

Reply:
I think that should work.

Reply:
@clintoncwolfe I think we can keep the required. We are able to separate our validation implementation for the audit log from the existing validate_option method, and we separated the audit log options from the current default_options so that those can be handled separately. This I think will make this change less breaking.